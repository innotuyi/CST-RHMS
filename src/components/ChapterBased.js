import React, { useState } from "react";
import { Row, Col, Container, Form, Button, Jumbotron } from "react-bootstrap";
import Header from "./Header";
import axios from 'axios';
import { useHistory } from "react-router-dom";

const ChapterBased = () => {
  const history = useHistory();
  const [title, setTitle] = useState("");
  const [author, setAuthor] = useState("");
  const [booktitle, setBooktitle] = useState("");
  const [chapterpage, setChapterPage] = useState("");
  const [place, setPlace] = useState("");
  const [editor, setEditor] = useState("");
  const [noofpages, setNofPages] = useState("");
  const [isbn, setISBN] = useState("");

  const submitHandler = (e) => {
    e.preventDefault();
    const payload = {
      'chapter_title': title, 
       'author':author,
       'book_title':booktitle,
       'chap_pages':chapterpage,
       'place':place,
       'editor':editor,   
       'no_of_pages':noofpages,
       'ISBN':isbn,       
       'research':'PRJA1'}


    axios.post('http://127.0.0.1:8000/chapter/', payload)
.then(function (response) {
console.log('Successful response: ',  response);
})
.catch(function (error) {
console.log('Error response: ', error);
});

history.push('/Chapter');
console.warn(
  title,
  author,
  chapterpage,
  place,
  editor,
  noofpages,
  isbn
);
};

  return (
    <div
      style={{
        background: "linear-gradient(to bottom, white, 84%, #1089AF)",
      }}
    >
      <Header />
      <Container>
        <Row className="d-flex justify-content-center arighn-item-center">
          <Col className="d-flex justify-content-center">
            <Jumbotron
              style={{
                background: "linear-gradient(to bottom, white , 80%, #1089AF )",
                borderRadius: "10px",
                width: "90%",
              }}
            >
              <Form className="form" style={{}} onSubmit={submitHandler}>
                <h2 className="text-center mb-5 text-secondary">
                  Book Based Information
                </h2>
                <Form.Group>
                  <Form.Label>Title</Form.Label>
                  <Form.Control
                    required
                    type="text"
                    id="title"
                    value={title}
                    onChange={(e) => setTitle(e.target.value)}
                    placeholder="Enter Chapter Title"
                  />
                </Form.Group>
                <Form.Group>
                  <Form.Label>Author</Form.Label>
                  <Form.Control
                    required
                    type="text"
                    id="author"
                    value={author}
                    onChange={(e) => setAuthor(e.target.value)}
                    placeholder="Enter Author's Name"
                  />
                </Form.Group>
                <Form.Group>
                  <Form.Label>Book Title</Form.Label>
                  <Form.Control
                    required
                    type="text"
                    id="booktitle"
                    value={booktitle}
                    onChange={(e) => setBooktitle(e.target.value)}
                    placeholder="Book Title"
                  />
                </Form.Group>
                <Form.Group>
                  <Form.Label>Editor</Form.Label>
                  <Form.Control
                    required
                    type="text"
                    id="editor"
                    value={editor}
                    onChange={(e) => setEditor(e.target.value)}
                    placeholder="Book Editor"
                  />
                </Form.Group>
                
                <Form.Group>
                  <Form.Label>chapter pages</Form.Label>
                  <Form.Control
                    required
                    type="number"
                    id="chapterpage"
                    value={chapterpage}
                    onChange={(e) => setChapterPage(e.target.value)}
                    placeholder="Book Title"
                  />
                </Form.Group>
                <Row>
                  {/* <Col>
                    <Form.Group controlId="dob">
                      <Form.Label>Select Publication Date</Form.Label>
                      <Form.Control
                        type="date"
                        name="dos"
                        id="pubdate"
                        value={pubDate}
                        onChange={(e) => setPubDate(e.target.value)}
                        placeholder="Publication Date"
                      />
                    </Form.Group>
                  </Col> */}
                  <Col>
                    <Form.Group>
                      <Form.Label>Publication Place</Form.Label>
                      <Form.Control
                        type="text"
                        name="dos"
                        id="place"
                        value={place}
                        onChange={(e) => setPlace(e.target.value)}
                        placeholder="Enter Publication Place"
                      />
                    </Form.Group>
                  </Col>
                </Row>
                <Row>
                  <Col>
                    <Form.Group controlId="">
                      <Form.Label>Number Of Pages</Form.Label>
                      <Form.Control
                        type="Number"
                        name=""
                        id="pugnumber"
                        value={noofpages}
                        onChange={(e) => setNofPages(e.target.value)}
                        placeholder="Enter The Number Of Pages"
                      />
                    </Form.Group>
                  </Col>
                  <Col>
                    <Form.Group>
                      <Form.Label>ISBN</Form.Label>
                      <Form.Control
                        type="Number"
                        name=""
                        id="isbn"
                        value={isbn}
                        onChange={(e) => setISBN(e.target.value)}
                        placeholder="Enter ISBN"
                      />
                    </Form.Group>
                  </Col>
                </Row>
                <Col className="text-center">
                  <Button className="" type="submit">
                    Proceed
                  </Button>
                </Col>
              </Form>
            </Jumbotron>
          </Col>
        </Row>
      </Container>
    </div>
  );
};
export default ChapterBased;
