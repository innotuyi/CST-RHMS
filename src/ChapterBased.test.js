import React from "react";
import ChapterBased from "./components/ChapterBased";

import { configure, shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

let wrapper;
const name = "Rwigamba";

beforeEach(() => {
  wrapper = shallow(<ChapterBased />);
});

describe("Chapter Based Tests", () => {
  it("matches Snapshot Collectly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

   it("should have All Inputs", () => {
     expect(wrapper.find("#author")).toHaveLength(1);
     expect(wrapper.find("#title")).toHaveLength(1);
     expect(wrapper.find("#chapter")).toHaveLength(1);
     expect(wrapper.find("#pubdate")).toHaveLength(1);
     expect(wrapper.find("#pubplace")).toHaveLength(1);
     expect(wrapper.find("#pagnumber")).toHaveLength(1);
     expect(wrapper.find("#isbn")).toHaveLength(1);
   });

   it("should have a Button  Component", () => {
     expect(wrapper.find("Button")).toHaveLength(1);
     expect(wrapper.find("Button").text()).toEqual("Create");
   });

   it("Should capture Author's Name correctly on Change", () => {
     wrapper.find("#author").simulate("change", {
       target: {
         value: name,
       },
     });
     expect(wrapper.find("#author").props().value).toEqual(name);
   });

   it("Should capture Book Title correctly on Change", () => {
     wrapper.find("#title").simulate("change", {
       target: {
         value: name,
       },
     });
     expect(wrapper.find("#title").props().value).toEqual(name);
   });

   it("Should capture Publication Date correctly on Change", () => {
     wrapper.find("#pubdate").simulate("change", {
       target: {
         value: name,
       },
     });
     expect(wrapper.find("#pubdate").props().value).toEqual(name);
   });

   it("Should capture Publication Place correctly on Change", () => {
     wrapper.find("#pubplace").simulate("change", {
       target: {
         value: name,
       },
     });
     expect(wrapper.find("#pubplace").props().value).toEqual(name);
   });

   it("Should capture Publication Date correctly on Change", () => {
     wrapper.find("#pubdate").simulate("change", {
       target: {
         value: "2021-06-16",
       },
     });
     expect(wrapper.find("#pubdate").props().value).toEqual("2021-06-16");
   });

   it("Should capture Page Number correctly on Change", () => {
     wrapper.find("#pagnumber").simulate("change", {
       target: {
         value: "20",
       },
     });
     expect(wrapper.find("#pagnumber").props().value).toEqual("20");
   });

   it("Should capture ISBN correctly on Change", () => {
     wrapper.find("#isbn").simulate("change", {
       target: {
         value: "20",
       },
     });
     expect(wrapper.find("#isbn").props().value).toEqual("20");
   });

});
