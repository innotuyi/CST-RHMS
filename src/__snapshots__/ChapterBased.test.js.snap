// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chapter Based Tests matches Snapshot Collectly 1`] = `
<div
  style={
    Object {
      "background": "linear-gradient(to bottom, white, 84%, #1089AF)",
    }
  }
>
  <Header />
  <Container
    fluid={false}
  >
    <Row
      className="d-flex justify-content-center arighn-item-center"
      noGutters={false}
    >
      <Col
        className="d-flex justify-content-center"
      >
        <Jumbotron
          fluid={false}
          style={
            Object {
              "background": "linear-gradient(to bottom, white , 80%, #1089AF )",
              "borderRadius": "10px",
              "width": "90%",
            }
          }
        >
          <Form
            className="form"
            inline={false}
            onSubmit={[Function]}
            style={Object {}}
          >
            <h2
              className="text-center mb-5 text-secondary"
            >
              Chapter Based Research Information
            </h2>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                Author name
              </FormLabel>
              <FormControl
                id="author"
                onChange={[Function]}
                placeholder="Enter The Name Of The Author"
                required={true}
                type="text"
                value=""
              />
            </FormGroup>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                Book Title
              </FormLabel>
              <FormControl
                id="title"
                onChange={[Function]}
                placeholder="Enter The Title Of The Book "
                required={true}
                type="text"
                value=""
              />
            </FormGroup>
            <FormGroup>
              <FormLabel
                column={false}
                srOnly={false}
              >
                Book Chapter
              </FormLabel>
              <FormControl
                id="chapter"
                onChange={[Function]}
                placeholder="Enter The Chapter Of The Book"
                required={true}
                type="text"
                value=""
              />
            </FormGroup>
            <Row
              noGutters={false}
            >
              <Col>
                <FormGroup
                  controlId="dob"
                >
                  <FormLabel
                    column={false}
                    srOnly={false}
                  >
                    Select Publication Date
                  </FormLabel>
                  <FormControl
                    id="pubdate"
                    name="dos"
                    onChange={[Function]}
                    placeholder="Publication Date"
                    type="date"
                    value=""
                  />
                </FormGroup>
              </Col>
              <Col>
                <FormGroup>
                  <FormLabel
                    column={false}
                    srOnly={false}
                  >
                     Place
                  </FormLabel>
                  <FormControl
                    id="pubplace"
                    name="dos"
                    onChange={[Function]}
                    placeholder="Enter Conference Location"
                    type="text"
                    value=""
                  />
                </FormGroup>
              </Col>
            </Row>
            <Row
              noGutters={false}
            >
              <Col>
                <FormGroup
                  controlId=""
                >
                  <FormLabel
                    column={false}
                    srOnly={false}
                  >
                    Chapter Pages
                  </FormLabel>
                  <FormControl
                    id="pagnumber"
                    name=""
                    onChange={[Function]}
                    placeholder="Enter The Number Of Pages"
                    type="Number"
                    value=""
                  />
                </FormGroup>
              </Col>
              <Col>
                <FormGroup>
                  <FormLabel
                    column={false}
                    srOnly={false}
                  >
                    ISBN
                  </FormLabel>
                  <FormControl
                    id="isbn"
                    name=""
                    onChange={[Function]}
                    placeholder="Enter ISBN"
                    type="Number"
                    value=""
                  />
                </FormGroup>
              </Col>
            </Row>
            <Col
              className="text-center"
            >
              <Button
                active={false}
                className=""
                disabled={false}
                type="submit"
                variant="primary"
              >
                Create
              </Button>
            </Col>
          </Form>
        </Jumbotron>
      </Col>
    </Row>
  </Container>
</div>
`;
